### PR TITLE
refactor: Use KIND to run integration tests

### PR DESCRIPTION
**Pull request by danielpacak**
The main changes:
- Use [KinD](https://github.com/marketplace/actions/kind-kubernetes-in-docker-action) to spawn a Kubernetes cluster in our CI for integration tests. Previously it was Rancher K3S run with Docker Compose
- Label [Cluster]Roles and [Cluster]RoleBindings with "app.kubernetes.io/name=who-can" to distinguish them from RBAC defaults in KinD. This will allow us to eventually have more reliable asserts in the integration tests.

Signed-off-by: Daniel Pacak <pacak.daniel@gmail.com>

### CODE REVIEW

File: .github/workflows/release.yaml [hunk 1]
```diff
@@ -14,15 +14,21 @@ jobs:
           go-version: 1.14
       - name: Checkout code
         uses: actions/checkout@v2
-      - name: Unshallow # This step is required for the changelog to work correctly
-        run: git fetch --prune --unshallow
+        with:
```
Comment (danielpacak) [inline .github/workflows/release.yaml:17]: This has changed in gorelease@v2 to properly generate changeslog from git commit messages

File: Makefile [hunk 2]
```diff
@@ -6,10 +6,8 @@ build: kubectl-who-can
...
 
-tests: $(SOURCES)
+unit-tests: $(SOURCES)
 	GO111MODULE=on go test -v -short -race -timeout 30s -coverprofile=coverage.txt -covermode=atomic ./...
 
 integration-tests: $(SOURCES)
-	GO111MODULE=on GOOS=linux CGO_ENABLED=0 go test -v -c -o test/bin/integration_test test/integration_test.go
-	@docker-compose -f test/docker-compose.yaml build --no-cache --force-rm
-	@docker-compose -f test/docker-compose.yaml up --abort-on-container-exit --exit-code-from test
+	GO111MODULE=on go test -v test/integration_test.go
```
Comment (danielpacak) [inline Makefile:13]: An integration test in pure Go thanks to Kind GitHub Action

File: test/integration_test.go [hunk 3]
```diff
@@ -103,6 +111,12 @@ func TestIntegration(t *testing.T) {
 			prettyPrintWhoCanOutput(t, tt.args, out)
 
 			for _, line := range tt.output {
+				// TODO Improve asserts on the output
```
Comment (danielpacak) [inline test/integration_test.go:114]: We need a separate PR for that. In this PR I just wanted to experiment with KinD GitHub Action.

File: test/integration_test.go [hunk 4]
```diff
@@ -13,20 +19,21 @@ import (
...
-	"strings"
-	"testing"
-	"time"
+)
+
+var (
+	// commonLabels is a set of common labels added to each object created by this integration test, which allows us
+	// to do a proper cleanup and distinguish them from the default object created on cluster init.
+	commonLabels = labels.Set{
+		"app.kubernetes.io/name": "who-can",
```
Comment (danielpacak) [inline test/integration_test.go:28]: k delete ns,crd,role,rolebinding,clusterrole,clusterrolebindings -l app.kubernetes.io/name=who-can

File: test/integration_test.go [hunk 5]
```diff
@@ -227,59 +266,86 @@ func configureRBAC(t *testing.T, coreClient client.Interface) {
...
 	_, err = clientRBAC.Roles(core.NamespaceDefault).Create(&rbac.Role{
-		ObjectMeta: meta.ObjectMeta{Name: "scale-workloads"},
+		ObjectMeta: meta.ObjectMeta{
+			Name:   "scale-workloads",
+			Labels: commonLabels,
+		},
 		Rules: []rbac.PolicyRule{
 			{
-				APIGroups: []string{"extensions"},
+				APIGroups: []string{"apps"}, // TODO For old clusters it extensions / never are apps
```
Comment (lizrice) [inline test/integration_test.go:311]: nit: I think there is a typo in this comment, is it something like "For old clusters it's extensions, newer are apps"?

File: test/integration_test.go (hunk 4 above)
Comment (lizrice) [reply test/integration_test.go:28]: would it make sense to add that command as a comment in the code?

File: test/integration_test.go (hunk 5 above)
Comment (danielpacak) [reply test/integration_test.go:311]: Well spotted. I was typing toooooo fast here :)

File: test/integration_test.go (hunk 4 above)
Comment (danielpacak) [reply test/integration_test.go:28]: Indeed. Can be useful for everyone trying to rerun the integration tests locally.

**Pull request merged**